assets/js/modules/ads/datastore/moduleData.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * `modules/ads` data store: module data.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies.
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ADS } from './constants';
+import { controls } from '../../../googlesitekit/datastore/site/info';
+const { createRegistrySelector } = Data;
+
+function getModuleDataProperty( propName ) {
+	return createRegistrySelector( ( select ) => () => {
+		const moduleData = select( MODULES_ADS ).getModuleData() || [];
+		return moduleData[ propName ];
+	} );
+}
```

Comment:
I thought I mentioned this in a review but maybe it was elsewhere… I'm a bit unclear on why this needs to be its own function rather than inlined into the selector, but I guess it's not a big deal 😅